portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,23 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function getComments() {
+  fetch('/data').then(response => response.json()).then((comments) => {
+    console.log(comments);
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+    const commentsListElement = document.getElementById('comments-section');
+    commentsListElement.innerHTML = '';
+    
+    for (com in comments) {
```

Comment:
fun way to do this:

comments.forEach(com => commentsListElement.appendChild(createListElement(comments[com])));

Also, why does the list element need to be created from comments[com] instead of just com?

Reply:
When I originally created the list elements with 'com' directly, it was only printing out the integer indices of the comments and not the comments themselves. As a result, I figured I would reference the comments by calling 'comments[com]'.

I just updated it to be the line of code you suggested and the list elements only needs to reference 'com'. I suppose this is just because of the different structures of the loops?

Reply:
forEach seems to pull the items in the JSON directly while for( _ in _ ) seems to give an integer iteration of the items in the JSON

Reply:
nice!